src/czbenchmarks/tasks/task.py

```diff
@@ -21,6 +29,294 @@ class TaskOutput(BaseModel):
     model_config = {"arbitrary_types_allowed": True}
 
 
+class TaskParameter(BaseModel):
+    """Schema for a single, discoverable parameter."""
+
+    type: str
+    default: Any = None
+    required: bool
+
+
+class TaskInfo(BaseModel):
+    """Schema for all discoverable information about a single benchmark task."""
+
+    name: str
+    display_name: str
+    description: str
+    task_params: Dict[str, TaskParameter]
+    baseline_params: Dict[str, TaskParameter]
+    metrics: List[str]
+
+
+class TaskRegistry:
+    """A registry that is populated automatically as Task subclasses are defined."""
+
+    def __init__(self):
+        self._registry: Dict[str, Type["Task"]] = {}
+        self._info: Dict[str, TaskInfo] = {}
+
+    def register_task(self, task_class: Type["Task"]):
+        """Registers a task class and introspects it to gather metadata."""
+        if inspect.isabstract(task_class) or not hasattr(task_class, "display_name"):
+            return
```

Comment:
Should this be an error? Or maybe at least log an error? Otherwise someone might be confused as to why their new task isn't showing up when they forgot to set a display name